infra/main.bicep

```diff
@@ -0,0 +1,117 @@
+targetScope = 'subscription'
+
+@minLength(1)
+@maxLength(64)
+@description('Name of the the environment which is used to generate a short unique hash used in all resources.')
+param environmentName string
+
+@minLength(1)
+@description('Primary location for all resources (filtered on available regions for Azure Open AI Service).')
+@allowed([ 'westeurope', 'southcentralus', 'australiaeast', 'canadaeast', 'eastus', 'eastus2', 'francecentral', 'japaneast', 'northcentralus', 'swedencentral', 'switzerlandnorth', 'uksouth' ])
+param location string
+
+//Leave blank to use default naming conventions
+param resourceGroupName string = ''
+param identityName string = ''
+param apimServiceName string = ''
+param logAnalyticsName string = ''
+param applicationInsightsDashboardName string = ''
+param applicationInsightsName string = ''
+
+param openAiInstances object = {
+  openAi1: {
+    name: 'openai1'
+    location: 'eastus'
+  }
+  openAi2: {
+    name: 'openai2'
+    location: 'northcentralus'
+  }
+  openAi3: {
+    name: 'openai3'
+    location: 'eastus2'
+  }
+}
+
+param chatGptModelVersion string = '0613'
+
+var abbrs = loadJsonContent('./abbreviations.json')
+var resourceToken = toLower(uniqueString(subscription().id, environmentName, location))
+var openAiSkuName = 'S0'
```

Comment:
Personally, I think it would be super helpful to have all the vars that the user might update in one consolidated group with comments around it.